WordPress/gutenberg
# Refactor convert links

## What?


Refactors the convert links functionality between the toolbar and panel and makes them consistent with one another.

## Why?


In preparation for #46469, I wanted to do some refactoring so the convert to blocks functionality only needed to be changed in one place.

I chose to use the 100 page limit from the toolbar 'Edit' button and the select block functionality from the side panel `Customize` button.

The limit, according to #30390, is because performance begins to significantly degrade after about that many pages. I think in a follow-up the UX should change to show a disabled button in both scenarios with an explanation for why it is disabled rather than just hiding it.

The select parent block was added in #46352 to improve on the UX, and I think that change should be brought over to the toolbar 'Edit' button and modal for consistency.

## How?


## Testing Instructions





1. Create a navigation block. (Make sure you have less than 100 pages on your site.)
2. Convert to links with the toolbar 'Edit' button and ensure that it behaves as expected.
3. Convert to links with the side panel 'Customize' button and ensure that it behaves as expected.
4. Add enough pages to surpass the 100 page limit (`wp post generate --count=100 --post_type=page --post_title='test'`)
5. See that the toolbar 'Edit' and side panel 'Customize' no longer show.

### Testing Instructions for Keyboard


## Screenshots or screencast
